src/clients/c++/library/request_grpc.cc

```diff
@@ -944,31 +927,32 @@ InferGrpcContextImpl::AsyncTransfer()
       if (this->exiting_) {
         return true;
       }
-      // wake up if at least one request is not ready
-      for (auto& ongoing_async_request : this->ongoing_async_requests_) {
-        if (std::static_pointer_cast<GrpcRequestImpl>(
-                ongoing_async_request.second)
-                ->IsReady() == false) {
-          return true;
-        }
+      // wake up if an async request has been generated
+      if (!this->ongoing_async_requests_.empty()) {
+        return true;
```

Comment:
can just be `return !this->ongoing_async_requests_.empty()`

Reply:
done